protocol/v2/sync/handlers/decided_history.go

```diff
@@ -38,7 +40,18 @@ func HistoryHandler(plogger *zap.Logger, storeMap *storage.QBFTStores, reporting
 			if store == nil {
 				return nil, errors.New(fmt.Sprintf("not storage found for type %s", msgID.GetRoleType().String()))
 			}
-			results, err := store.GetDecided(msgID[:], sm.Params.Height[0], sm.Params.Height[1])
+			states, err := store.GetInstance(msgID[:], sm.Params.Height[0], sm.Params.Height[1])
+			results := make([]*specqbft.SignedMessage, 0, len(states))
+			for _, state := range states {
+				_, msgs := state.CommitContainer.LongestUniqueSignersForRoundAndValue(state.Round, state.DecidedValue)
```

Comment:
Please use - 
LastPreparedRound instead of just round

Reply:
done